crates/router/src/core/payments/helpers.rs

```diff
@@ -1252,7 +1252,8 @@ pub(crate) fn validate_pm_or_token_given(
             Some(api_enums::PaymentMethodType::Paypal)
         ) && !matches!(mandate_type, Some(api::MandateTxnType::RecurringMandateTxn))
             && token.is_none()
-            && (payment_method_data.is_none() || payment_method.is_none()),
+            && (payment_method_data.is_none() || payment_method.is_none())
+            && !matches!(payment_method, Some(api_enums::PaymentMethod::BankRedirect)),
```

Comment:
why there is a change here? aren't you getting payment_method_data and payment_method